src/agent/rustjail/src/validator.rs

```diff
@@ -4,17 +4,15 @@
 //
 
 use crate::container::Config;
-use anyhow::{anyhow, Context, Error, Result};
+use anyhow::{anyhow, Context, Result};
 use oci::{Linux, LinuxIdMapping, LinuxNamespace, Spec};
 use std::collections::HashMap;
 use std::path::{Component, PathBuf};
 
-fn einval() -> Error {
-    anyhow!(nix::Error::EINVAL)
-}
-
 fn get_linux(oci: &Spec) -> Result<&Linux> {
-    oci.linux.as_ref().ok_or_else(einval)
+    oci.linux
+        .as_ref()
+        .ok_or_else(|| anyhow!("Unable to get Linux from Spec"))
```

Comment:
Nit: This makes sense to developers but might be clearer for end users as something like:

```suggestion
        .ok_or_else(|| anyhow!("Unable to get Linux section from Spec"))
```

But, either way, this message is inconsistent with the (redundant) same check on the Linux spec in `src/agent/rustjail/src/container.rs`.